kolibri/logger/models.py

```diff
@@ -98,3 +100,56 @@ class UserSessionLog(BaseLogModel):
     start_timestamp = models.DateTimeField(auto_now_add=True)
     completion_timestamp = models.DateTimeField(blank=True, null=True)
     pages = models.TextField(blank=True)
+
+class MasteryLog(BaseLogModel):
+    """
+    This model provides a summary of a user's engagement with an assessment within a mastery level
+    """
+    # Every MasteryLog is related to the single summary log for the user/content pair
+    summarylog = models.ForeignKey(ContentSummaryLog, related_name="masterylogs")
+    # The MasteryLog records the mastery criterion that has been specified for the user.
+    # It is recorded here to prevent this changing in the middle of a user's engagement
+    # with an assessment.
+    mastery_criterion = models.CharField(max_length=200)
+    start_timestamp = models.DateTimeField()
+    end_timestamp = models.DateTimeField(blank=True, null=True)
+    completion_timestamp = models.DateTimeField(blank=True, null=True)
+    # The integer mastery level that this log is tracking.
+    mastery_level = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)])
+    # Has this mastery level been completed?
+    complete = models.BooleanField(default=False)
+
+    user_field = "summarylog__user"
+
+    def infer_dataset(self):
+        return self.summarylog.dataset
+
+class AttemptLog(BaseLogModel):
+    """
+    This model provides a summary of a user's engagement within a particular interaction with an
+    item in an assessment
+    """
+    # Unique identifier within the relevant assessment for the particular question/item
+    # that this attemptlog is a record of an interaction with.
+    item = models.CharField(max_length=200)
+    start_timestamp = models.DateTimeField()
+    end_timestamp = models.DateTimeField(blank=True, null=True)
+    completion_timestamp = models.DateTimeField(blank=True, null=True)
+    # Which mastery log was this attemptlog associated with?
+    masterylog = models.ForeignKey(MasteryLog, related_name="attemptlogs")
+    time_spent = models.FloatField(help_text="(in seconds)", default=0.0, validators=[MinValueValidator(0)])
+    complete = models.BooleanField(default=False)
+    # How correct was their answer? In simple cases, just 0 or 1.
+    correct = models.FloatField(validators=[MinValueValidator(0), MaxValueValidator(1)])
+    # JSON blob that would allow the learner's answer to be rerendered in the frontend interface
+    answer = models.TextField()
+    # A human readable answer that could be rendered directly in coach reports, can be blank.
+    simple_answer = models.CharField(max_length=200, blank=True)
+    # A JSON Array with a sequence of JSON objects that describe the history of interaction of the user
+    # with this assessment item in this attempt.
+    interaction_history = models.TextField()
```

Comment:
This piece is something we (meaning @jayoshih et al) will need to spec out in more detail, since the schema doesn't enforce any particular format here.